dnf/cli/output.py

```diff
@@ -2250,3 +2261,119 @@ def progressbar(current, total, name=None):
         sys.stdout.write('\n')
 
     sys.stdout.flush()
+
+class PackageFormatter(object):
```

Comment:
I don't agree with the name (especially since you are proposing to add it to API). "formatters" should be initialized with a format, not with the object that should be formatted. (Like in [logging](https://docs.python.org/dev/howto/logging.html#logging.logging.Formatter.__init__))